source/hp/dof_handler.cc

```diff
@@ -1425,51 +1446,29 @@ namespace hp
         {
           saved_subdomain_ids.resize(shared_tria->n_active_cells());
 
-          typename parallel::shared::Triangulation<dim, spacedim>::
-            active_cell_iterator cell = get_triangulation().begin_active(),
-                                 endc = get_triangulation().end();
-
           const std::vector<types::subdomain_id> &true_subdomain_ids =
             shared_tria->get_true_subdomain_ids_of_cells();
 
-          for (unsigned int index = 0; cell != endc; ++cell, ++index)
+          for (const auto &cell : shared_tria->active_cell_iterators())
             {
+              const unsigned int index   = cell->active_cell_index();
               saved_subdomain_ids[index] = cell->subdomain_id();
               cell->set_subdomain_id(true_subdomain_ids[index]);
             }
         }
 
-    // ensure that the active_fe_indices vectors are
-    // initialized correctly.
-    create_active_fe_table();
```

Comment:
why is this block no longer needed?

Reply:
I moved that block to the `set_fe()` function, that will be called earlier in `distribute_dofs()`.

It didn't appear to me that this section requires the correct subdomains to be set in the `p::s`  case. I may be wrong. We may also just leave this block of code in.

Reply:
We could move the `set_fe()` call from above to here. I guess this would make it more obvious.